apps/cli/src/wallet.ts

```diff
@@ -8,80 +8,146 @@ import {
     HttpTransport,
     PublicClient,
     WalletClient,
+    custom,
+    Account,
 } from "viem";
-import { mnemonicToAccount } from "viem/accounts";
+import { mnemonicToAccount, privateKeyToAccount } from "viem/accounts";
+import { MetaMaskSDK } from "@metamask/sdk";
+import { EthereumProvider } from "@walletconnect/ethereum-provider";
+import qrcode from "qrcode-terminal";
+import ansiColors from "ansi-colors";
+import {
+    arbitrum,
+    arbitrumGoerli,
+    foundry,
+    mainnet,
+    optimism,
+    optimismGoerli,
+    sepolia,
+} from "@wagmi/chains";
+
+import { Choice, hexInput, selectAuto } from "./prompts.js";
+
+// list of all chains that will eventually be supported
+export const chains: Chain[] = [
+    arbitrum,
+    arbitrumGoerli,
+    mainnet,
+    optimism,
+    optimismGoerli,
+    sepolia,
+];
+
+// list of chains that are currently enabled
+const enabledChains: Chain[] = [arbitrumGoerli, optimismGoerli, sepolia];
 
-const DEFAULT_DEVNET_MNEMONIC =
+// build a select choice from a Chain object
+const chainChoices = chains.map((chain: Chain) => ({
+    name: chain.name,
+    value: chain,
+    disabled:
+        enabledChains.findIndex((c) => c.id === chain.id) == -1
+            ? "(coming soon)"
+            : false,
+}));
+
+export const DEFAULT_DEVNET_MNEMONIC =
     "test test test test test test test test test test test junk";
 
+export type WalletType =
+    | "metamask"
+    | "walletconnect"
+    | "mnemonic"
+    | "private-key";
+const walletChoices = (chain: Chain): Choice<WalletType>[] => {
```

Comment:
I could only test the mnemonic option.

Should we already have an AWS KMS option?